src/Chainweb/Pact/PactService.hs

```diff
@@ -273,12 +268,27 @@ initPactService' ver cid chainwebLogger bhDb pdb sqlenv config act = do
                 , _psCheckpointerDepth = 0
                 }
         !pst = PactServiceState Nothing mempty initialParentHeader P.noSPVSupport
-    runPactServiceM pst pse act
+    runPactServiceM pst pse $ do
+
+        -- If the latest header that is stored in the checkpointer was on an
+        -- orphaned fork, there is no way to recover it in the call of
+        -- 'initalPayloadState.readContracts'. We therefore rewind to the latest
+        -- avaliable header in the block header database.
+        --
+        initializeLatestBlock
+        act
   where
     initialBlockState = initBlockState $ genesisHeight ver cid
     loggers = pactLoggers chainwebLogger
     logger = P.newLogger loggers $ P.LogName ("PactService" <> show cid)
 
+initializeLatestBlock :: PayloadCasLookup cas => PactServiceM cas ()
+initializeLatestBlock = findLatestValidBlock >>= \case
+    Nothing -> return ()
+    Just b -> withBatch $ rewindTo initialRewindLimit (Just $ ParentHeader b)
+  where
+    initialRewindLimit = Just 1000
```

Comment:
What happens when this is exceeded at startup?

Reply:
Good point. I added `exitOnRewindLimitExceeded` that makes it terminate the process and exit with a meaningful error message. It's the same what already happens when `execValidateBlock` hits the rewind limit.